### PR TITLE
Implementation of getSearchCondition() in VField with Exposed

### PR DESCRIPTION
**Pull request by ZMarouani**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 1]
```diff
@@ -656,43 +687,118 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
...
         VConstants.SOP_EQ -> null
-        VConstants.SOP_NE -> "IS NOT NULL"
-        else -> "IS NULL"
+        VConstants.SOP_NE -> (nullCondBuilder {
+          IsNotNullOp(it)
+        } to null)
+        else -> (nullCondBuilder {
+          IsNullOp(it)
+        } to null)
       }
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:711]: here you return nullCondBuilder with type  (Expression<String>.(t: String) -> Op<Boolean>) but your method should return Pair<Expression<String>.(t: String) -> Op<Boolean>, Any?>? so check it !

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 2]
```diff
@@ -656,43 +687,118 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
...
+          GreaterOp(it, it.wrap(operand))
+        }
+        "<=" -> operatorBuilder {
+          LessEqOp(it, it.wrap(operand))
+        }
+        ">=" -> operatorBuilder {
+          GreaterEqOp(it, it.wrap(operand))
+        }
+        "<>" -> operatorBuilder {
+          NeqOp(it, it.wrap(operand))
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:759]: Here your operator receives a function that contains the operand and then at the end of the function you return a pair of operator & operand?? Check it!

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 3]
```diff
@@ -656,43 +687,118 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
...
       }
-      when (options and VConstants.FDO_SEARCH_MASK) {
-        VConstants.FDO_SEARCH_NONE -> {
-        }
-        VConstants.FDO_SEARCH_UPPER -> operand = "{fn UPPER($operand)}"
-        VConstants.FDO_SEARCH_LOWER -> operand = "{fn LOWER($operand)}"
-        else -> throw InconsistencyException("FATAL ERROR: bad search code: $options")
-      }
-      "$operator $operand"
+      return ((operator as Expression<String>.(t: String) -> Op<Boolean>) to (operand as TypeVariable<GenericDeclaration>?))
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:801]: can you check your cast !

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 4]
```diff
@@ -656,43 +687,118 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
...
   open fun getSearchCondition(): String? {
-    return if (isNull(block!!.activeRecord)) {
-      when(getSearchOperator()) {
+    TODO()
+  }
+
+  /**
+   * Returns the search conditions for this field.
+   */
+  open fun getSearchCondition_(): Pair<Expression<String>.(t: String) -> Op<Boolean>, Any?>? {
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:696]: I will send to you a proposition by mail

**Comment by hfazai** [comment]
Closed in favor to #152

**Pull request closed without merging**